### PR TITLE
feat(ui): adds Sentry project selector field

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a sentry project selector field (`sentry_project_selector`) where the user can pick a project from a select field that shows the project icon. This is going to be used in the upcoming stack trace mapping UI. Here's what it looks like in that form:

Empty state:
Screen Shot 2020-10-23 at 11 25 17 AM

Dropdown state:
Screen Shot 2020-10-23 at 11 25 24 AM

Selected state:
Screen Shot 2020-10-23 at 11 25 30 AM

Note that I was thinking of using the `AvatarProject` project type since that's all that's needed to render the avatar. However, I want the selected value of the dropdown to be the id instead of the slug and the id field is missing there. I could use something like `AvatarProject & {id: string}` but I'm not sure how useful that would be.

### CODE REVIEW

**Comment by scefali** [comment]
I realize I need to make more changes for this to work properly

**Comment by scefali** [comment]
fixed

File: src/sentry/static/sentry/app/views/settings/components/forms/sentryProjectSelectorField.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+class RenderField extends React.Component<RenderProps> {
+  static defaultProps = defaultProps;
+
+  //need to map the option object to the value
+  handleChange = (
+    onBlur: Props['onBlur'],
+    onChange: Props['onChange'],
+    optionObj: {value: any},
+    event: any
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/sentryProjectSelectorField.tsx:30]: This could be `React.MouseEvent` 🤷

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**